### PR TITLE
Rewrite Managing User Information section

### PR DESCRIPTION
**Pull request by mspensieri**
Part 2 of a multi-part effort to enhance the "Users" section of our
documentation. This part tackles the "Managing User Information" page,
and clarifies the difference between structured and unstructured
fields, cleans up the code samples to be more readable and direct, and
adds a mention of how and when user properties are stored and uploaded
on each of our SDKs.

managing-user-information

Javascript got cut off in the previous screenshot... here it is

managing-user-information-javascript

### CODE REVIEW

File: pages/guide/managing-user-information.md [hunk 1]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
-curl https://api.smooch.io/v1/apps/5963c0d619a30a2e00de36b8/appusers/c7f6e6d6c3a637261bd9656f \
-     -X PUT \
-     -d '{"givenName": "Hilda", "properties": {"your-custom-property": "some-value"}}' \
-     -H 'content-type: application/json' \
-     -H 'authorization: your-account-jwt'
-```
+## Structured Fields
 
-## Adding properties using the SDKs
+Structured fields are properties that Smooch has identified as common across many use cases, and has chosen to expose in a structured way. The currently supported structured fields are:
```
Comment (spasiu) [inline pages/guide/managing-user-information.md]: Perhaps I would change _"... and has chosen to expose in a structured way."_ to _"... and has exposed as common properties across all users, when present."_

Comment (spasiu) [reply pages/guide/managing-user-information.md]: "a structured way" seemed ambiguous, and "chosen" seems arbitrary.

File: pages/guide/managing-user-information.md [hunk 2]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
 
-Here, we add a property and name from the Web SDK:
+- `givenName`, also referred to as `firstName` in some contexts, which represents the user's given name
+- `surname`, also referred to as `lastName` in some contexts, which represents the user's surname
+- `signedUpAt`, which is a date when the user first started using your service, or when they first became a customer. If not customized, this field is automatically populated to be the first time the user was created in Smooch, which is most likely the moment when the user messaged you for the first time.
```
Comment (spasiu) [inline pages/guide/managing-user-information.md]: "...which is a date when..." -> "which is **the** date when"

File: pages/guide/managing-user-information.md [hunk 3]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
+In addition to the information automatically collected and stored for each of a user's clients, an [appUser](/guide/intro-to-users/) itself can have metadata and profile information attached to it, in order to better understand the context and the history of the user.
 
-Custom properties can be added to a user:
-1. via the [REST API](http://docs.smooch.io/rest/#update-app-user)
-2. at runtime with the mobile and Web SDKs
+Profile information can be added to a user in two ways:
 
-In addition to custom properties, there are a number of predefined properties such as `givenName`, `surname`, and `email` that can also be set via the SDKs, or REST API.
+- [Via the REST API](/guide/managing-user-information/#adding-properties-using-the-rest-api)
+- [At runtime with the mobile and wb SDKs](/guide/managing-user-information/#adding-properties-using-the-sdks)
```
Comment (jpjoyal) [inline pages/guide/managing-user-information.md]: wb->web

File: pages/guide/managing-user-information.md [hunk 4]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
 
-## Adding properties using the SDKs
+Structured fields are properties that Smooch has identified as common across many use cases, and has exposed as common properties across all users, when present. The currently supported structured fields are:
 
-Here, we add a property and name from the Web SDK:
+- `givenName`, also referred to as `firstName` in some contexts, which represents the user's given name
+- `surname`, also referred to as `lastName` in some contexts, which represents the user's surname
+- `signedUpAt`, which is the date when the user first started using your service, or when they first became a customer. If not customized, this field is automatically populated to be the date the user was created in Smooch, which is most likely the moment when the user messaged you for the first time.
+- `email`, which represents the user's email address. Note that this field is only for _display purposes_, and does not affect how Smooch delivers messages in any way. For information on how to link an email address to a user to be able to send them messages over email, see the [business initiated conversations](/guide/business-initiated-conversations/#email) guide.
```
Comment (jpjoyal) [inline pages/guide/managing-user-information.md]: is only -> is only used

File: pages/guide/managing-user-information.md [hunk 5]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
 
-![Email updated in Slack](/images/email_updated.png)
+On Android and iOS, when a user property is set using one of the SDK methods, the properties are immediately serialized to disk until they can be uploaded to the server. Changes to user properties are uploaded in batches at regular intervals while the app is in the foreground, as well as just before the app is sent to the background, or immediately before a message is sent by the user. If the application exits unexpectedly, or the user has intermittent internet connection or no internet connection, the properties will remain on disk until the upload eventually succeeds (even across app launches). If the user does not yet exist (i.e. the user has not yet sent their first message, and the [startConversation](/guide/business-initiated-conversations/) method has not been called), the properties are still tracked and stored on disk until the user is eventually created, and they will be uploaded as part of the user creation flow.
```
Comment (jpjoyal) [inline pages/guide/managing-user-information.md]: 👏

File: pages/guide/managing-user-information.md [hunk 6]
```diff
@@ -6,225 +6,159 @@ layout: two-column
...
+}
 ```
 
-### iOS
-
-The `addProperties:` method accepts an `NSDictionary` containing the properties to add. This dictionary must have keys that are `NSString` and values that are either `NSString`, `NSNumber` or `NSDate`. If your dictionary contains any other data type as a value, then `description` will be called on the object and the resulting NSString will be added as a property.
+## Unstructured Fields
 
-### Android
+Unstructured fields, also referred to as "custom properties", are a set of key/value pairs that are specific to your application domain, or use case. These fields are stored under the `properties` field of an appUser, and can have values of type `Number`, `String`, or `Boolean`.
```
Comment (jpjoyal) [inline pages/guide/managing-user-information.md]: Would add our size limit notice here too.

Comment (mspensieri) [reply pages/guide/managing-user-information.md]: It's just below in the `<aside>`

**Pull request merged**